webapp/src/serial.tsx

```diff
@@ -74,12 +76,17 @@ export class Editor extends srceditor.Editor {
         const data = smsg.data || ""
         const source = smsg.id || "?"
 
+        if (!(source in this.sourceMap)) {
+            this.sourceMap[source] = lf("source") + (this.sourceIdx++).toString()
```

Comment:
the app theme has a "boardName" variable that would read good (e.g. micro:bit ...). Typically, we simply do ``!!this.sourceMap[source]`` for lookups.
sourceIdx = Object.keys(this.sourceMap).length